src/bindgen.rs

```diff
@@ -34,40 +61,112 @@ pub fn wasm_bindgen_build(
     name: &str,
     disable_dts: bool,
     target: &str,
+    install_permitted: bool,
     debug: bool,
     step: &Step,
 ) -> Result<(), Error> {
     let msg = format!("{}Running WASM-bindgen...", emoji::RUNNER);
     PBAR.step(step, &msg);
     let binary_name = name.replace("-", "_");
     let release_or_debug = if debug { "debug" } else { "release" };
-    let wasm_path = format!(
-        "target/wasm32-unknown-unknown/{}/{}.wasm",
-        release_or_debug, binary_name
-    );
-    let dts_arg = if disable_dts == false {
-        "--typescript"
+
+    if let Some(wasm_bindgen_path_str) = wasm_bindgen_path_str(path, install_permitted)? {
+        let wasm_path = format!(
+            "target/wasm32-unknown-unknown/{}/{}.wasm",
+            release_or_debug, binary_name
+        );
+
+        let dts_arg = if disable_dts == false {
+            "--typescript"
+        } else {
+            "--no-typescript"
+        };
+
+        let target_arg = match target {
+            "nodejs" => "--nodejs",
+            _ => "--browser",
+        };
+
+        let bindgen_path = path::Path::new(&wasm_bindgen_path_str);
+        let output = Command::new(bindgen_path)
+            .current_dir(path)
+            .arg(&wasm_path)
+            .arg("--out-dir")
+            .arg("./pkg")
+            .arg(dts_arg)
+            .arg(target_arg)
+            .output()?;
+        if !output.status.success() {
+            let s = String::from_utf8_lossy(&output.stderr);
+            Err(Error::cli("wasm-bindgen failed to execute properly", s))
+        } else {
+            Ok(())
+        }
     } else {
-        "--no-typescript"
-    };
+        Err(Error::crate_config("Could not find `wasm-bindgen`"))
+    }
+}
 
-    let target_arg = match target {
-        "nodejs" => "--nodejs",
-        _ => "--browser",
-    };
+/// Check if the `wasm-bindgen` dependency is locally satisfied.
+fn wasm_bindgen_version_check(bindgen_path: &str, dep_version: &str) -> Result<bool, Error> {
+    let wasm_bindgen = path::Path::new(bindgen_path);
 
-    let output = Command::new("wasm-bindgen")
-        .current_dir(path)
-        .arg(&wasm_path)
-        .arg("--out-dir")
-        .arg("./pkg")
-        .arg(dts_arg)
-        .arg(target_arg)
-        .output()?;
-    if !output.status.success() {
+    let output = Command::new(wasm_bindgen).arg("--version").output()?;
+    if output.status.success() {
+        let s = String::from_utf8_lossy(&output.stdout);
+        let installed_version = s.trim();
+        Ok(installed_version == dep_version)
```

Comment:
Running `wasm-bindgen --version` doesn't produce *just* the version number:

```
$ wasm-bindgen --version
wasm-bindgen 0.2.15
```

So I think we need to do a little bit more parsing here than just `.trim()`, since we are getting the expected version from `Cargo.toml` which *will* just be the version number.

Reply:
Ah! Good catch. I can take care of that.